Presentation/MasterDetailSelection.swift

```diff
@@ -156,7 +162,7 @@ public extension MasterDetailSelection {
 
             immediate = true
             let presentDisposable = vc.present(presentation.onDismiss {
-                if self.isCollapsed.value && !immediate {
+                if isCollapsed && !immediate {
```

Comment:
Is this changing the behavior as you using an old value in an async op? 
If not perhaps `isCollapsed` should be name `wasCollapsed`

Reply:
great catch, I missed the fact that this block of code is async. I'm struggling to decide what to do in the nil case here because I think that's not gonna happen (see a bit above we have an early return when isCollapsed is nil and it will never change from true/false to nil). I'm just gonna return early in that case. Btw this is the behaviour that this code influences and it was broken with the previous commit!
![splitvc-deselect](https://user-images.githubusercontent.com/1555713/65423577-affa5000-de09-11e9-9307-027700425ff8.gif)

Reply:
Added an assertion failure since that will notify us if this behaviour is not as we expect it in debug builds.